guides/source/active_record_callbacks.md

```diff
@@ -134,16 +168,148 @@ Here is a list with all the available Active Record callbacks, listed in the sam
+[`around_create`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Callbacks/ClassMethods.html#method-i-around_create
+[`around_save`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Callbacks/ClassMethods.html#method-i-around_save
+[`before_create`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Callbacks/ClassMethods.html#method-i-before_create
+[`before_save`]:
+    https://api.rubyonrails.org/classes/ActiveRecord/Callbacks/ClassMethods.html#method-i-before_save
+[`before_validation`]:
+    https://api.rubyonrails.org/classes/ActiveModel/Validations/Callbacks/ClassMethods.html#method-i-before_validation
+
+There are examples below that show how to use these callbacks. We've grouped
+them by the operation they are associated with, however they can be used in any
+combination.
+
+#### Validation Callbacks
+
+Validation callbacks are triggered by the `valid?` method. They are called
+before and after the validation phase.
+
+```ruby
+class User < ApplicationRecord
+  validates :name, presence: true
+  before_validation :normalize_name
+  after_validation :check_errors
+
+  private
+
+  def normalize_name
+    self.name = name.downcase.titleize if name.present?
+    Rails.logger.info("Name normalized to #{name}")
+  end
+
+  def check_errors
+    if errors.any?
+      Rails.logger.error("Validation failed: #{errors.full_messages.join(', ')}")
+    end
+  end
+end
+```
+
+```irb
+irb> user = User.new(name: "", email: "john.doe@me.com", password: "abc123456")
+=> #<User id: nil, email: "john.doe@me.com", created_at: nil, updated_at: nil, name: "">
+
+irb> user.valid?
+Name normalized to
+Validation failed: Name can't be blank
+=> false
+```
+
+#### Save Callbacks
+
+Save callbacks are triggered by the `save` method. They are called before, after
+and around the object is saved.
+
+```ruby
+class User < ApplicationRecord
+  before_save :encrypt_password
+  around_save :log_saving
+  after_save :send_welcome_email
+
+  private
+
+  def encrypt_password
+    self.password = BCrypt::Password.create(password)
+    Rails.logger.info("Password encrypted for user with email: #{email}")
+  end
+
+  def log_saving
+    Rails.logger.info("Saving user with email: #{email}")
+    yield
+    Rails.logger.info("User saved with email: #{email}")
+  end
+
+  def update_cache
+    Rails.cache.write("user_data", attributes)
+    Rails.logger.info("Update Cache")
+  end
+end
+```
+
+```irb
+irb> user = User.create(name: "Jane Doe", email: "jane.doe@gmail.com")
+
+Password encrypted for user with email: jane.doe@gmail.com
+Saving user with email: jane.doe@gmail.com
+User saved with email: jane.doe@gmail.com
+Update Cache
+=> #<User id: 1, email: "jane.doe@gmail.com", created_at: "2024-03-20 16:02:43.685500000 +0000", updated_at: "2024-03-20 16:02:43.685500000 +0000", name: "Jane Doe">
+```
+
+#### Create Callbacks
+
+Create callbacks are triggered by the `create` method. They are called before,
+after and around the object is created.
```

Comment:
Also similar to validate/save callbacks, this can be triggered by `create`, `save`, and also `update` (on a new / non-persisted record)

```suggestion
Create callbacks are triggered whenever the record is persisted (i.e. "saved")
to the underlying database **for the first time**, in other words, when we're
saving a new record, via the `create`, `save`, or `update` methods. They are
called before, after and around the object is created.
```

Reply:
✅